support-e2e/tests/test/checkout.ts

```diff
@@ -1,17 +1,52 @@
-import { expect, test } from '@playwright/test';
+import { expect, Page, test } from '@playwright/test';
 import { email, firstName, lastName } from '../utils/users';
 import { setupPage } from '../utils/page';
-import { setTestUserRequiredDetails } from '../utils/testUserDetails';
+import {
+	setTestUserDetails,
+	setTestUserRequiredDetails,
+} from '../utils/testUserDetails';
 import { fillInPayPalDetails } from '../utils/paypal';
 import { fillInCardDetails } from '../utils/cardDetails';
 import { checkRecaptcha } from '../utils/recaptcha';
+import { TestFields, ukWithPostalAddressOnly } from '../utils/userFields';
 
+// TODO: it'd be great to make the types here more specific, possibly using the
+// shared types from the product catalog.
 type TestDetails = {
 	product: string;
 	ratePlan: string;
 	paymentType: string;
 	internationalisationId: string;
-	paymentFrequency: string;
+};
+
+const setUserDetailsForProduct = async (
```

Comment:
I introduced this method because we now need different user details depending on the product/country (i.e. Tier Three needs full address data). In future it might be nice to do this in a more generic way, e.g. with some kind of mapping, but I think this'll do for now.